### ISSUE TITLE
[BUG] Trakt "Your Next Episodes" not showing correct episode after reseting watch history on Trakt

### ISSUE THREAD

**Issue by notlot** [jurialmunkey/plugin.video.themoviedb.helper#1077]
### Describe the bug

With a premium trakt account, I reset the watch history I had for The Witcher, so my 'next episode' was s01e01 instead of s03e01, which is the next unwatched episode. On the TMDB helper, when I go to the "Your Next Episodes" page, it still presents s03e01 as the next Witcher episode to watch. Although it is at the front of the list as I started to rewatch season 1 episodes, which suggests to me that it knows its the latest seen show, just not showing the correct episode according to my Trakt account. I also noticed that if I go to the "Your In-Progress TV shows" page and open the show, it correctly lists s01e01 as the first episode to watch and has been updating as I continue to rewatch the episodes (I am now on s02e04). Presumably the two lists are being populated by different methods but I'm not sure. 

This behavour is the same across my laptop, desktop, and android box w/ CoreElec, which are running the same custom build I have for AH2 -> TMDBhelper -> Seren. Let me know if you need other info. 

Thanks in advance!

### Steps To Reproduce

1. Open TMDB Helper addon
2. Go to Trakt
3. Go to "Your Next Episodes"
 - wrong episode listed here

### Checklist

- [X] I have searched the issues and this bug has not been reported.
- [X] I have updated TMDbHelper to the most recent version.
- [X] I have described the bug and included steps to reproduce the issue
- [ ] I have uploaded my debug log.

**Comment by jurialmunkey** [comment]
Caching issue on Trakt server.

See here:
https://www.github.com/trakt/api-help/issues/32#issuecomment-468468608

Some trakt end points just take longer to refresh. Nothing I can do about it.

**Comment by notlot** [comment]
Ahh I see, thanks! I also found that the same list in Seren shows the 'wrong' episode as well, so my bad for starting a bug issue. 

It's been almost a week now and the "Your Next Episodes" list is still showing the first unwatched episode, not respecting the trakt rewatch progress. Do you think there's something I could do on my end or is this just a trakt issue that we live with?

**Comment by jurialmunkey** [comment]
A week is definitely too long - so it's not going to be caching issue after all.

Looking into it a bit more: From what I'm reading on the Trakt support github it sounds like the VIP feature to reset progress the Trakt API doesn't actually update the next_episode field in the sync for the show progress. The Trakt server updates next_episode in every other scenario except this one.

I saw a weird response from kalehulk (Trakt dev) on reddit that the client app should check every episode in the show and compare against the reset date -- seems weird to me to have a next_episode field if sometimes it is not the next episode...

**Comment by jurialmunkey** [comment]
To add to the above, the reason in-progress does show the correct behaviour is because it does the long way of checking against the reset_at date -- but that's also slower to calculate.

**Comment by jurialmunkey** [comment]
@notlot - ~Can you test this version and confirm that the workaround solves the problem?~

EDIT: Scratch that. I've added a better approach to the main branch. 
Can you test 5.1.23 (latest release) and confirm that it fixes the issue?

**Comment by notlot** [comment]
Confirmed that the issue is fixed on ver. 5.1.23. 

Cheers!

**Comment by brentsmith721** [comment]
@jurialmunkey I am on version 5.1.26 but having a similar issue. As described in the original bug, after using the Trakt "Rewatch" feature I do not see the correct episode on the "Your Next Episodes" page. Additionally, while the original bug reported that the "Your In-Progress TV shows" page did show the correct episode, in my case it does not. Both the "Your Next Episodes" and "Your In-Progress TV shows" pages show my original watch progress and are not respecting the reset.

I tried this for 2 shows. One show is still airing, and my original watch progress had unwatched episodes. For that show, the "Your Next Episodes" and "Your In-Progress TV shows" pages display the next episode and the show but with my original watch progress. However, for the other show I tried this with, the show is no longer airing and I had already watched 100% of the show. And in this case it is not showing up at all on either the "Your Next Episodes" or "Your In-Progress TV shows" page.

![screenshot00000](https://github.com/jurialmunkey/plugin.video.themoviedb.helper/assets/42626892/70e11ace-5fef-40b0-9c7b-858f411df1f5)
![screenshot00001](https://github.com/jurialmunkey/plugin.video.themoviedb.helper/assets/42626892/bdf3ea1f-65d0-4772-ab32-b20e4a67e363)
<img width="675" alt="Screenshot 2023-07-30 at 6 03 10 PM" src="https://github.com/jurialmunkey/plugin.video.themoviedb.helper/assets/42626892/a0f627cf-d2ba-4381-98d6-6229a7001792">
